sdk/signalr/azure-messaging-webpubsubservice/setup.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+
+import os
+import re
+
+from setuptools import setup, find_packages
+
+
+# Change the PACKAGE_NAME only to change folder and different name
+PACKAGE_NAME = "azure-messaging-webpubsubservice"
+PACKAGE_PPRINT_NAME = "Azure WebPubSub Service"
+
+# a-b-c => a/b/c
+package_folder_path = PACKAGE_NAME.replace("-", "/")
+
+
+# Version extraction inspired from 'requests'
+with open(os.path.join(package_folder_path, "_version.py"), "r") as fd:
+    version = re.search(
+        r'^VERSION\s*=\s*[\'"]([^\'"]*)[\'"]', fd.read(), re.MULTILINE
+    ).group(1)
+
+if not version:
+    raise RuntimeError("Cannot find version information")
+
+setup(
+    name=PACKAGE_NAME,
+    version=version,
+    description="Microsoft {} Client Library for Python".format(PACKAGE_PPRINT_NAME),
+    long_description=open("README.md", "r").read(),
+    long_description_content_type="text/markdown",
+    license="MIT License",
+    author="Microsoft Corporation",
+    author_email="azpysdkhelp@microsoft.com",
+    url="https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/signalr/azure-messaging-webpubsubservice",
+    classifiers=[
+        "Development Status :: 5 - Production/Stable",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 2",
+        "Programming Language :: Python :: 2.7",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "License :: OSI Approved :: MIT License",
+    ],
+    zip_safe=False,
+    packages=find_packages(
+        exclude=[
+            # Exclude packages that will be covered by PEP420 or nspkg
+            "azure",
+            "azure.messaging",
+            "tests",
+            "examples",
+        ]
+    ),
+    install_requires=[
+        "azure-core<2.0.0,>=1.10.0",
+        "msrest>=0.6.18",
+        "cryptography>=2.1.4",
+        "pyjwt>=1.7.1",
```

Comment:
We need to upgrade the version of `pyjwt` to 2.0.x

Reply:
Related issue https://github.com/jpadilla/pyjwt/issues/505

Reply:
Upgrading to pyjwt 2.0.x drops python 2.7 support. Looking at the issue referenced, I'm not sure why we need it?

Reply:
The current version return `byte` instead of `string` in `jwt.encode()` in python3, which makes the generated token incorrect. If you don't want to upgrade jwt version, you need to fix it here with a `decode('UTF-8')`.
https://github.com/Azure/azure-sdk-for-python/pull/18199/files#diff-418e1fbe16ab098ec7fe81f33d656dbca5ce789bd35f01dcca74754bae5b1215R89

Reply:
six to the rescue. Fixed.